deploy/charts/monitoring/dashboards/node-inventory.json

```diff
@@ -0,0 +1,386 @@
+  "editable": true,
+  "gnetId": null,
+  "graphTooltip": 0,
+  "id": 4,
+  "links": [],
+  "panels": [
+    {
+      "datasource": null,
+      "description": "Based on rox_compliance_call_node_inventory_duration_seconds. Aggregated over 4h windows.",
+      "fieldConfig": {
+        "defaults": {
+          "color": {
+            "mode": "thresholds"
+          },
+          "mappings": [],
+          "thresholds": {
+            "mode": "absolute",
+            "steps": [
+              {
+                "color": "green",
+                "value": null
+              },
+              {
+                "color": "red",
+                "value": 80
+              }
+            ]
+          },
+          "unit": "s"
+        },
+        "overrides": []
+      },
+      "gridPos": {
+        "h": 9,
+        "w": 12,
+        "x": 0,
+        "y": 0
+      },
+      "id": 20,
+      "options": {
+        "displayMode": "gradient",
+        "orientation": "horizontal",
+        "reduceOptions": {
+          "calcs": [
+            "lastNotNull"
+          ],
+          "fields": "",
+          "values": false
+        },
+        "showUnfilled": true,
+        "text": {}
+      },
+      "pluginVersion": "7.5.15",
+      "targets": [
+        {
+          "exemplar": true,
+          "expr": "rate(rox_compliance_call_node_inventory_duration_seconds_sum[4h])/rate(rox_compliance_call_node_inventory_duration_seconds_count[4h])",
+          "interval": "",
+          "legendFormat": "{{exported_node_name}}",
+          "refId": "A"
+        }
+      ],
+      "title": "Average Node Scan Duration",
+      "type": "bargauge"
+    },
+    {
+      "datasource": null,
+      "description": "Based on rox_compliance_protobuf_inventory_message_size_bytes. Aggregated over 4h windows.",
+      "fieldConfig": {
+        "defaults": {
+          "color": {
+            "mode": "thresholds"
+          },
+          "decimals": 2,
+          "mappings": [],
+          "thresholds": {
+            "mode": "absolute",
+            "steps": [
+              {
+                "color": "green",
+                "value": null
+              }
+            ]
+          },
+          "unit": "decbytes"
+        },
+        "overrides": []
+      },
+      "gridPos": {
+        "h": 9,
+        "w": 12,
+        "x": 12,
+        "y": 0
+      },
+      "id": 15,
+      "options": {
+        "colorMode": "value",
+        "graphMode": "none",
+        "justifyMode": "auto",
+        "orientation": "horizontal",
+        "reduceOptions": {
+          "calcs": [
+            "lastNotNull"
+          ],
+          "fields": "",
+          "values": false
+        },
+        "text": {},
+        "textMode": "auto"
+      },
+      "pluginVersion": "7.5.15",
+      "targets": [
+        {
+          "exemplar": true,
+          "expr": "rate(rox_compliance_protobuf_inventory_message_size_bytes_sum[4h])/rate(rox_compliance_protobuf_inventory_message_size_bytes_count[4h])",
```

Comment:
I am not fully sure about the `rate` function here, but it's been a while since I used it. Have you validated that the returned numbers are realistic?

Reply:
The numbers line up with my observations from development time of this feature, yes.
With regards to the `rate` function:
I adapted the example of the Prometheus Docs detailing how to calculate averages from Histograms ([here](https://prometheus.io/docs/practices/histograms/#count-and-sum-of-observations)).